crypto/x509/x509_def.c

```diff
@@ -8,27 +8,125 @@
  */
 
 #include <stdio.h>
+#include "internal/e_os.h"
 #include "internal/cryptlib.h"
 #include <openssl/crypto.h>
 #include <openssl/x509.h>
 
+#if defined(_WIN32)
+static char *x509_private_dir = NULL;
+static char *x509_cert_area = NULL;
+static char *x509_cert_dir = NULL;
+static char *x509_cert_file = NULL;
+static char *openssldir = NULL;
```

Comment:
Don't we leak all these values if they get set?

Also don't we need to lock access to them? Maybe via a `RUN_ONCE`